core/dbt/contracts/graph/unparsed.py

```diff
@@ -93,6 +93,8 @@ class HasDocs(AdditionalPropertiesMixin, ExtensibleDbtClassMixin, Replaceable):
     description: str = ""
     meta: Dict[str, Any] = field(default_factory=dict)
     data_type: Optional[str] = None
+    constraints: Optional[List[str]] = None
+    constraints_check: Optional[str] = None
```

Comment:
Yes, just postgres + databricks. We'll include adapter-specific disclaimers in this PR: https://github.com/dbt-labs/docs.getdbt.com/pull/2601

Reply:
_[For follow-up issues, out of scope for this PR]_

This didn't occur to me on the first read-through, but is there a reason why we want this to be a separate configuration, rather than just a subset of the available `constraints` (on platforms that support it)?

Answering my own question: It looks like it's because Databricks requires that `check` constraints have to be applied _after_ table creation via `alter` statements. (It looks like Postgres supports them within the `create table` statement.)

I still think I might prefer a configuration like:
```python
constraints: Optional[List[Union[str, Dict[str, str]]]] = None
```
```yml
columns:
  - name: price
    data_type: numeric
    constraints:
      - not null
      - check: price > 0
        name: positive_price  # Postgres supports naming constraints
```

Postgres + Databricks also support table-level constraints that are composed of multiple columns. We should also support `constraints` at the model level, in addition to the column level. Should they be a config, or a non-config attribute?